File: october data structure/count.cpp
#include <bits/stdc++.h>
using namespace std;
int MAX = 20;

vector<int> phi(MAX);
vector<int> primes;
 
 

int main(){
    for(int p = 1; p < MAX; p++)
        phi[p] = p;

    for(int p = 2; p < MAX; p++){
        if (phi[p] == p){
            primes.push_back(p);
            
            phi[p] = p - 1;
    
            for(int i = p * 2; i < MAX; i += p)
                phi[i] = (phi[i] / p) * (p - 1);
        }
    }
    
    int cases, N;
    cin >> cases;
    for(int tc = 0; tc < cases; tc++){

        cin >> N;

        int countOut = N - phi[N], divisors = 1;

        for(auto p: primes){
            int count = 0;
            while (N % p == 0){
                N = N / p;
                count += 1;
            }   
            divisors *= count + 1;
        }
        cout << countOut - divisors + 1 << '\n';
    }
    
}
 
  

File: october data structure/count.py
MAX = 10**6 + 6
 
phi = [i for i in range(MAX)]
primes = []
 
 
for p in range(2, MAX):
    
    if (phi[p] == p): 
        primes.append(p)
        
        phi[p] = p - 1
 
        for i in range(2 * p, MAX, p): 
            phi[i] = (phi[i] // p) * (p - 1) 
 
for tc in range(int(input())):
    N = int(input())
    
    countOut = N - phi[N]
 
    divisors = 1
 
    for p in primes:
        count = 0
        if N % p == 0:
            while N % p == 0:
                N = N // p
                count += 1
            divisors *= count + 1

    print(countOut - divisors + 1)
 
  